### PR TITLE
Quickstart/snowflake

### PR DESCRIPTION
**Pull request by rikturr**
Snowflake quickstart . "Advanced" folder was taken from examples-gpu/nyc-taxi-snowflake

If someone else ends up merging this, please don't delete the `quickstart/snowflake` branch. This branch is currently being used for the quickstarts in the product until we swap over to auto-creating them for this repo

### CODE REVIEW

**Comment by jameslamb** [comment]
> If someone else ends up merging this, please don't delete the `quickstart/snowflake` branch. This branch is currently being used for the quickstarts in the product until we swap over to auto-creating them for this repo

@rikturr are you an admin on this repo? If you are and find yourself in one of these "I need to preserve a long-lived feature branch" situations, you can get better automated protection against that by setting up a branch protection.

1. Go to https://github.com/saturncloud/examples/settings/branches
2. Click "add rule"
3. Add a rule that leaves "allow deletions" unchecked, and check the "include administrators" box to be sure other admins don't delete your branch

image

4. Click "Create"

I just did that for this branch, so it'll be safe until you're ready to remove it.

File: examples/snowflake/README.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+This quickstart gives you what you need to get started connecting to Snowflake from Saturn Cloud. This includes:
+
+* Creating a connection to a Snowflake database to pull data into a Pandas DataFrame
+* Querying Snowflake data over a distributed cluster using Dask for when the data is too big to fit into memory on a single computer. _Once you have the data loaded in a distributed setting you can use tools like RAPIDS or PyTorch to train models on it, as can be seen in other Quickstarts._
+
+Both of these tasks can be found in the [snowflake-dask.ipynb](snowflake-dask.ipynb) Jupyter notebook.
+
+## _Advanced_ - NYC Taxi analysis
+
+Once you're familiar with how to query data from Snowflake, check out the [advanced](advanced/README.md) folder for complete examples of a classification exercise with NYC Taxi data. Those notebooks compare the runtime of a CPU-based pipeline with a GPU-based one to illustrate the massive speedups you can get when pairing Snowflake with GPUs on Saturn Cloud!
```
Comment (jameslamb) [inline examples/snowflake/README.md]:
```suggestion
Once you're familiar with how to query data from Snowflake, check out the [advanced](./advanced/README.md) folder for complete examples of a classification exercise with NYC Taxi data. Those notebooks compare the runtime of a CPU-based pipeline with a GPU-based one to illustrate the massive speedups you can get when pairing Snowflake with GPUs on Saturn Cloud!
```

I've found that for some markdown renderers, relative links like this only work if you explicitly prefix them with `./`. I can't remember if it was GitHub or the "Markdown Preview" in JupyterLab.

File: examples/snowflake/advanced/README.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+### Random forest classification
+
+The random forest notebooks are an example of GPU-acceleration for model training with [RAPIDS](http://rapids.ai/). The [`cuml` package](https://github.com/rapidsai/cuml) features an API that closely resembles scikit-learn, but does all processing on GPUs. This leads to some crazy performance speedups! What if your data doesn't fit into a single GPU's memory? That's where Dask comes in! RAPIDS has native Dask integrations to be able to run on a cluster of machines with GPUs. Buckle up, this is going to be a fun ride!
+
+Run these notebooks in order:
+
+1. [`rf-scikit.ipynb`](rf-scikit.ipynb): single-node scikit version - takes a while 🙁
+1. [`rf-rapids.ipynb`](rf-rapids.ipynb): single-GPU RAPIDS version - super fast! 🤯
+1. [`rf-rapids-dask.ipynb`](rf-rapids-dask.ipynb): multi-GPU Dask+RAPIDS version - _extra_ super fast! 🤯🤯🤯
```
Comment (jameslamb) [inline examples/snowflake/advanced/README.md]:
```suggestion
1. [`rf-scikit.ipynb`](./rf-scikit.ipynb): single-node scikit version - takes a while 🙁
1. [`rf-rapids.ipynb`](./rf-rapids.ipynb): single-GPU RAPIDS version - super fast! 🤯
1. [`rf-rapids-dask.ipynb`](./rf-rapids-dask.ipynb): multi-GPU Dask+RAPIDS version - _extra_ super fast! 🤯🤯🤯
```

**Review by jameslamb — COMMENTED** [review]
left a few small comments. I'll go to ReviewNB next.

**Review by jameslamb — COMMENTED** [review]
Ok I looked on ReviewNB, all seems find to me! I didn't run any of the code though. I'm also leaving just a "comment" review since I think it should probably be the data science team deciding if this gets merged or not.

If you need to me to run the code or you need a quick approval, feel free to `@` me.

File: examples/snowflake/.saturn/saturn.json [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+        "worker_size": "g4dnxlarge"
+    },
+    "environment_variables": {
+        "TAXI_DATABASE": "saturn_nyc_taxi",
+        "JUPYTERLAB_SATURN_EXTENSION_DISABLED": "True"
+    },
+    "description": "Connect to a Snowflake database from either a single machine or distributed clsuter",
+    "title": "Snowflake",
+    "thumbnail_image_url": "https://saturn-public-assets.s3.us-east-2.amazonaws.com/example-thumbnails/snowflake.png",
+    "weight": 2
```
Comment (jsignell) [inline examples/snowflake/.saturn/saturn.json]: I am going to open a PR to enforce uniqueness among weights. So it'd be good to have them not all be 2 :). I think this one should be fairly high. Maybe 10 for instance.

File: examples/snowflake/.saturn/saturn.json [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    "dask_cluster": {
+        "n_workers": 3,
+        "nthreads": 1,
+        "scheduler_size": "medium",
+        "worker_size": "g4dnxlarge"
+    },
+    "environment_variables": {
+        "TAXI_DATABASE": "saturn_nyc_taxi",
+        "JUPYTERLAB_SATURN_EXTENSION_DISABLED": "True"
+    },
```
Comment (jsignell) [inline examples/snowflake/.saturn/saturn.json:16]: There is a notion in examples-cpu and examples-gpu of `required_credentials`. It'd be good to have a list of those here.

File: examples/snowflake/.saturn/saturn.json (hunk 3 above)
Comment (rikturr) [reply examples/snowflake/.saturn/saturn.json]: I actually didn't realize how weights worked - I thought it was more a rank order thing and ties would be chosen some other way. I'll update :)

Comment (jsignell) [reply examples/snowflake/.saturn/saturn.json]: No that is how they work. But we should know what order we want for these things and set them intentionally.

**Review by jnolis — CHANGES_REQUESTED** [review]
This looks good! The changes I have here are the same as the the prefect one. Can you:

* Ensure the titles all work if these were docs out of context (which I think unlike the prefect one is true here)
* Move images to the `saturn-public-assets` bucket so the refs will work when they are docs
* Make sure the first cell just has the title and nothing else (which I think is untrue for the advanced ones)

Thanks!

**Comment by rikturr** [comment]
tests passing: https://github.com/saturncloud/release-images/runs/2142770711?check_suite_focus=true 

Will work on the titles and images now

**Review by jsignell — APPROVED** [review]

**Review by jnolis — APPROVED** [review]

**Pull request merged**